manifests/tmpfile.pp

```diff
@@ -26,23 +26,24 @@
 #  * Mutually exclusive with ``$limits``
 #
 define systemd::tmpfile(
-  Enum['present', 'absent', 'file'] $ensure  = 'file',
-  Stdlib::Absolutepath              $path    = '/etc/tmpfiles.d',
-  Optional[String]                  $content = undef,
-  Optional[String]                  $source  = undef,
+  Enum['present', 'absent', 'file'] $ensure   = 'file',
+  Systemd::Dropin                   $filename = $name,
+  Stdlib::Absolutepath              $path     = '/etc/tmpfiles.d',
+  Optional[String]                  $content  = undef,
+  Optional[String]                  $source   = undef,
 ) {
   include systemd::tmpfiles
 
-  if $name =~ Pattern['/'] {
-    fail('$name may not contain a forward slash "(/)"')
+  if $filename =~ Pattern['/'] {
```

Comment:
This should work as well:
```puppet
if $name =~ Systemd::Dropin` {}
```
You don't need to introduce an additional parameter.

Reply:
`Systemd::Dropin` does not include the not no / in it.

As for additional parameter I was just copying the style of `systemd::dropin_file`

Advantage of  making it a parameter is the error is a lot more obvious I would say when you hit it.

The not containing a `/` should be added to the `Systemd::Dropin` type of course.

Reply:
ah thanks for pointing that out!